### PR TITLE
Fix "from:" search term working properly

### LINKED ISSUE #18791: `from:` term is not working properly for remote user
### Steps to reproduce the problem

1. search `from:localuser` on search bar
2. results are shown. (which are posts that I interacted)
3. search `from:user@fromremote.tld` on search bar
4. results are empty.
...

### Expected behaviour

Results are posts that I interacted before

### Actual behaviour

Nothing shown

### Specifications

Mastodon v3.5.3, v3.5.3+glitch
Browser: firefox

### PR DESCRIPTION
**Pull request by tribela**
This pr fixes search term like `from:username@remote.host`

This pr also fixes #18791

### CODE REVIEW

File: app/services/search_service.rb [hunk 1]
```diff
@@ -90,7 +90,7 @@ def url_resource_symbol
   def full_text_searchable?
     return false unless Chewy.enabled?
 
-    statuses_search? && !@account.nil? && !((@query.start_with?('#') || @query.include?('@')) && !@query.include?(' '))
+    statuses_search? && !@account.nil? && !((@query.start_with?('#') || @query.match(/^(@\w+|@?\w+@[\w.]+)$/)) && !@query.include?(' '))
```
Comment (Gargron) [inline app/services/search_service.rb:93]: I'd prefer to remove that check instead of having to maintain that regex.

**Review by Gargron — CHANGES_REQUESTED** [review]

File: app/services/search_service.rb (hunk 1 above)
Comment (tribela) [reply app/services/search_service.rb:93]: Completely remove regex and always do full text search?

**Comment by tribela** [comment]
Rebased onto new main branch, Removed regex.
Please review it again

**Comment by Gargron** [comment]
Fixed by #26673

**Pull request closed without merging**